src/project/components/giveTime/giveTime.js

```diff
@@ -81,52 +79,19 @@ GiveTimeDialog.propTypes = {
 
 const mapStateToProps = (state) => {
     return {
-        openId: state.project.giveTimeDialog.openId,
-        amount: state.project.giveTimeDialog.amount,
-        userCredit: state.project.giveTimeDialog.userCredit,
-        userId: state.project.user.id,
-        userRowId: state.project.user.rowId,
+        openId: state.project.giveTime.giveTimeDialog.openId,
+        amount: state.project.giveTime.giveTimeDialog.amount,
+        userCredit: state.project.login.user.credit,
+        userId: state.project.login.user.id,
+        userRowId: state.project.login.user.rowId,
     }
 }
 
 const mapDispatchToProps = (dispatch) => {
-    return {
-        onSubmit: (form) => {
-            dispatch(getGraphQL(`
-                mutation giveTime(
-                    $projectRowId: Int!,
-                    $userRowId: Int!,
-                    $credit: Int!
-                ){
-                    projectGiveTime(input: {
-                        personId: $userRowId,
-                        projectId: $projectRowId,
-                        amount: $credit
-                    }) {
-                        output {
-                            rowId,
-                            acquired
-                        }
-                    }
-                }`,
-                {
-                    credit: form.amount,
-                    userRowId: form.userRowId,
-                    projectRowId: form.projectRowId,
-                },
-                () => {
-                    dispatch(giveTime(form.amount, form.projectRowId))
-                    dispatch(closeGiveTimeDialog())
-                }
-            ))
-        },
-        openDialog: (id) => {
-            dispatch(openGiveTimeDialog(id))
-        },
-        closeDialog: () => {
-            dispatch(closeGiveTimeDialog())
-        },
-    }
+    return bindActionCreators({
+        onSubmit: actions.onSubmit,
```

Comment:
It is good! Was wondering if we should put a higher order function...with dispatch in the closure...
Like:
```javascript
bindActionCreatorsCustom(actions.onSubmit,actions.openDialog, actions.closeDialog)
```